src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -159,7 +159,7 @@ public QuorumServer(long id, String hostname,
         public void recreateSocketAddresses() {
             InetAddress address = null;
             try {
-                address = InetAddress.getByName(this.hostname);
+                address = getReachableAddress(this.hostname, 2000);
```

Comment:
Instead of hardcoding 2000 here, we can have a constant variable represent the timeout value at the beginning of QuorumPeer.java. [QuorumCnxManager](https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java) is a good example on how it uses constants.